code/game/machinery/computer/medical.dm

```diff
@@ -294,6 +297,12 @@
 							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))
 								return
 							src.active1.fields["age"] = t1
+					if("species")
+						if(active1)
+							var/t1 = input("Select a species", "Species Selection") as null|anything in roundstart_species
+							if ((!( t1 ) || !( src.authenticated ) || usr.stat || usr.restrained() || (!in_range(src, usr) && (!istype(usr, /mob/living/silicon))) || src.active1 != a1))

```

Comment:
Is it too much to ask to replace most of this with usr.canUseTopic(src) and then move the check outside these ifs? (and replace it with a switch or at least else-ifs?)